File: tests/machine/x/fs/sort_stable.fs
open System

type Anon1 = {
    n: int
    v: string
}
let items: obj list = [{ n = 1; v = "a" }; { n = 1; v = "b" }; { n = 2; v = "c" }]
let result: obj list = [ for i in items do yield i.v ] |> List.sortBy (fun i -> i.n)
printfn "%A" (result)

File: tests/transpiler/x/clj/sort_stable.clj
(ns main)

(require 'clojure.set)

(defrecord Anon1 [n v])

(def items [{:n 1 :v "a"} {:n 1 :v "b"} {:n 2 :v "c"}])

(def result (for [i (sort-by (fn [i] (:n i)) items)] (:v i)))

(defn -main []
  (println result))

(-main)
